.github/workflows/release.yml

```diff
@@ -0,0 +1,39 @@
+# References:
+# - https://github.com/cli/cli/blob/v2.32.1/.github/workflows/deployment.yml
+# - https://github.com/goreleaser/goreleaser-action/tree/v4.4.0#usage
+
+name: Release
+
+permissions:
+  contents: write
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag_name:
+        required: true
+        type: string
+      go_version:
+        default: "1.20"
+        type: string
+
+jobs:
+  goreleaser:
```

Comment:
The `jobs` name is `gorelease`? 🤔 Why not just `release`? 😁 
`goreleaser` is kinda an implementation detail and can/should be avoided IMHO.

Reply:
well for me the workflow name should be different from the job name. And we are releasing a go app so 'goreleaser' sounds okay to me. Another option: we name workflow as 'deploy' and then job name as 'release'. WDYT?

Reply:
or we can say `release_cli_app` :smile:

Reply:
Its naming 🤷 I don't care *that* much.
But I wouldn't go with `goreleaser` as this is an implementation detail.
A library/programm we use. And we should expose this.
But if you are fine with goreleaser, lets go with that 🙂